lib/device.js

```diff
@@ -251,6 +256,174 @@ const util = require('util'),
+
+                        capFilename = parseBase;
+                        outputDirPath = path.resolve(parseDirPath);
+                    } else if (parseBase && (parseExt === "")) {
+                        // it's directory path. Use it as destination dir Path
+                        capFilename = _createDateFileName();
+                        outputDirPath = path.resolve(options.inputPath);
+                    }
+                }
+
+                options.capFormat = capFormat;
+                options.capFilename = capFilename;
+                options.capDirPath = "/tmp/capture",
+                options.sourcePath = path.join(options.capDirPath, capFilename);
+                options.destinationPath = outputDirPath;
+                options.ignore = true;
+                options.silent = true;
+                
+                next();
+            }
+
+            function _createDateFileName() {
+                // Capture file name generate : targetname_displayX_yyyymmddhhmmss.png"
+                function __getDate(){
+                    const curDate = new Date();
+                    const dateFormat = __pad(curDate.getFullYear(), 2)
+                        +  __pad((curDate.getMonth()+1), 2)
+                        + __pad(curDate.getDate(), 2)
+                        + __pad(curDate.getHours(), 2)
+                        + __pad(curDate.getMinutes(), 2)
+                        + __pad(curDate.getSeconds(), 2);
+
+                    return dateFormat;
+                }
+
+                function __pad(number, length) {
+                    let str = '' + number;
+                    while (str.length < length) {
+                        str = '0' + str;
+                    }
+                    return str;
+                }
+
+                const fileName = options.session.target.name + "_" +"display" + options.display+ "_" + __getDate() +".png";
+                return fileName;
+            }
+
+            function _captureFile(next) {
+                log.info("device#capture#_captureFile()");
+                const cmd = "/bin/mkdir -p " + options.capDirPath;
+                options.session.run(cmd, null, null, null, function(err) {
+                    if (err) {
+                        return setImmediate(next, err);
+                    } else {
+                        options.targetCapDir = true;
+                        const target = options.session.getDevice(),
+                            addr = target.lunaAddr.captureCompositorOutput,
+                            param = {
+                                // luna param
+                                subscribe: false,
+                                output : options.sourcePath,
+                                format: options.capFormat,
+                                displayId : options.display
+                            };
+
+                        luna.sendWithoutErrorHandle(options, addr, param, function(lineObj, next) {
```

Comment:
why do you use sendWithoutErrorHandle?
if some error occurs, we don't check them.

Reply:
We can check "luna-send error handle" in luna.sendWithoutErrorHandle, too.
Please verify 
{code}
$ ./ares-device.js -c -dp 10
ares-device ERR! CLI: luna-send command failed <ERR_INVALID_DISPLAY> 
{code}

I understand that the luna.send is better than sendWithoutErrorHandle. Right?